main.go

```diff
@@ -28,6 +28,8 @@ func main() {
 		os.Exit(1)
 	}
 
+	isGenerateDummy, _ := strconv.ParseBool(os.Getenv("KSOPS_GENERATE_DUMMY_SECRETS"))
```

Comment:
This line technically works I guess, but is somewhat confusing.

Consider the case when KSOPS_GENERATE_DUMMY_SECRETS isn't defined in the env - then `Getenv` returns empty string as per go os package docs. `ParseBool` fails to parse empty string into a bool and returns an error, which is ignored here. `isGenerateDummy` gets the default value of false, which is correct behaviour, but you need to know all these things to understand that. Would be best to check the error here and assign false more explicitly.

Reply:
Before doing actual commit. Just want to confirm if the env vars doesn't set, it will still get false but then the "explicit" set of false is doing from the code?
Sample code 
```golang
        isGenerateDummy, err := strconv.ParseBool(os.Getenv("KSOPS_ENV"))

	if err != nil {
		_, _ = fmt.Fprintf(os.Stderr, "error converting string to Boolean: %q \n", err)
		os.Exit(1)
	}
```

Reply:
just to avoid the obligation to set the env vars especially on local development environment

Reply:
If we want to be more fail-safe I'm thinking it should be like this:
* If `KSOPS_GENERATE_DUMMY_SECRETS` is not defined in the env or is empty (same from go's perspective), then set `isGenerateDummy` to false and procceed. This the default production mode that will run in ArgoCD.
* If `KSOPS_GENERATE_DUMMY_SECRETS` is non-empty, but there's an error parsing it into a bool then there should be an error printed out and ksops should exit with non-zero return code.
* If `KSOPS_GENERATE_DUMMY_SECRETS` parses into a valid bool then that's the value to be used for `isGenerateDummy`